extensions/amp-auto-ads/0.1/ad-tracker.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {resourcesForDoc} from '../../../src/resources';
+
+export class AdTracker {
+
+  /**
+   * @param {!Array<!Element>} ads
+   */
+  constructor(ads) {
+    /** @type {!Array<!Element>} */
+    this.ads_ = ads.slice(0);
```

Comment:
I don't think the slice is necessary, since it'll always be a new array instance.